atomic_reactor/dirs.py

```diff
@@ -45,6 +46,7 @@ def __init__(self, path: Path, platform: str) -> None:
         self.path = path
         self.platform = platform
         self.exported_squashed_image: Path = self.path / EXPORTED_SQUASHED_IMAGE_NAME
+        self.build_result_file = self.path / "build_result.json"
```

Comment:
This feels wrong - the directory will never have both the `build_result_file` and an `exported_squashed_image`, `dockerfile`, etc. It will only have the `build_result_file` if it is used from a ContextDir, and it will only have the others if it's used from a RootBuildDir